Entities/Attachable.cpp

```diff
@@ -297,6 +306,35 @@ void Attachable::Detach()
 }
 
 
+/// <summary>
+/// Turns on/off this Attachable's collisions, by adding/removing its atoms to/from its parent's AtomGroup.
+/// </summary>
+/// <param name="enable">Adds this Attachable's atoms to the parent's AtomGroup if true, removes them if false.</param>
+void Attachable::EnableTerrainCollisions(bool enable)
+{
+	if (IsAttached() && IsSetToCollide())
```

Comment:
I don't quite understand why you need the IsSetToCollide stuff to avoid duplicate atom groups. I know that that stops users from trying to make things that would have wonky physics collide, but I'd like to understand why the dual atomgroup thing was happening and how this fixed it.

Also, I think m_IsSetToCollide should be renamed to m_IsAllowedToCollideWithTerrain or something that more clearly explains that it needs to be on to allow terrain collisions specifically.

Reply:
m_IsSetToCollide is saying "I'm capable of colliding with terrain".
m_CollidesWithTerrain was repurposed to be a flag that says "I'm currently colliding with terrain" or "I'm currently not colliding with terrain".

The multi-collisionbox generating was happening because there wasn't anything saying "I'm currently colliding with terrain, do not add my atoms to the parent atomgroup again".

This method basically looks like:

```
if (i have a parent and i'm capable of colliding with terrain)
    if (i'm currently not colliding with terrain and i'm supposed to)
        add my atoms to the parent atomgroup to enable collisions
        i'm currently colliding with terrain
    else
        if (i'm currently colliding with terrain and i'm not supposed to)
            remove my atoms from the parent atomgroup to disable collisions
            i'm currently not colliding with terrain
```

And yeah I probably should rename it
Maybe `m_CanCollideWithTerrain`?

Reply:
I think if we keep it it definitely needs to be renamed. m_CanCollideWithTerrain sounds fine to me.

That said, I don't quite get the need for it still, aside from the wonky physics if you turn terrain collisions on for something that didn't start with them (which I'm okay with having til we fix it). To use your breakdown, why would this not work:
```if (i have a parent) //Note the removal of the can collide with terrain check
    if (i'm currently not colliding with terrain and i'm supposed to)
        add my atoms to the parent atomgroup to enable collisions
        i'm currently colliding with terrain
    else
        if (i'm currently colliding with terrain and i'm not supposed to)
            remove my atoms from the parent atomgroup to disable collisions
            i'm currently not colliding with terrain```

Reply:
This will work but the restriction on lua EnableTerrainCollisions will be gone, meaning you will be able to enable/disable collisions for attachables that were not set to CollidesWithTerrain = true in ini leading to wonky physics.

The whole thing is 
```
if (i have a parent and i'm capable of colliding with terrain)
    if (i'm currently not colliding with terrain and i'm supposed to)
        add my atoms to the parent atomgroup to enable collisions
        i'm currently colliding with terrain
    else
        if (i'm currently colliding with terrain and i'm not supposed to)
            remove my atoms from the parent atomgroup to disable collisions
            i'm currently not colliding with terrain
else
    if (i have a parent and i'm not capable of colliding with terrain)
        if (i'm supposed to or i'm not supposed to)
            lolno because you're not capable of this
```

Reply:
Yeah, I'm okay with the wonky physics as long as we have an issue to fix it in future. Once it's fixed it'll be a lot more sensible than how it is now where if you want to have it off and then turn it on later you have to turn it on in ini then off in lua then on later in lua.